tests/subsys/fs/fat_fs_api/prj_reentrant.conf

```diff
@@ -0,0 +1,13 @@
+CONFIG_FILE_SYSTEM=y
+CONFIG_FILE_SYSTEM_MKFS=y
+CONFIG_LOG=y
+CONFIG_FAT_FILESYSTEM_ELM=y
+CONFIG_DISK_DRIVER_FLASH=y
+CONFIG_SPI=y
+CONFIG_GPIO=y
+CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y
+CONFIG_FLASH=y
+CONFIG_FLASH_MAP=y
+CONFIG_FS_FATFS_REENTRANT=y
+CONFIG_MULTITHREADING=y
```

Comment:
You do not have to make entire prj file to add these. The prj differs by two configs from the non reentrant one, so you can use `extra_config` option of testcases.yaml to run with original prj.conf.